web-marketplace/src/components/organisms/ProjectAccountSelector/ProjectAccountSelector.tsx

```diff
@@ -0,0 +1,138 @@
+import { useState } from 'react';
+import { msg } from '@lingui/core/macro';
+import { useLingui } from '@lingui/react';
+
+import CheckIcon from 'web-components/src/components/icons/CheckIcon';
+import DropdownIcon from 'web-components/src/components/icons/DropdownIcon';
+import { Body } from 'web-components/src/components/typography';
+import { Subtitle } from 'web-components/src/components/typography/Subtitle';
+import UserAvatar from 'web-components/src/components/user/UserAvatar';
+import { cn } from 'web-components/src/utils/styles/cn';
+
+import useClickOutside from '../../../utils/hooks/useClickOutside';
+import { AccountOption } from '../DashboardNavigation/DashboardNavigation.types';
+
+export interface ProjectAccountSelectorProps {
+  accounts: AccountOption[];
+  selectedAddress: string;
+  onSelect: (id: string) => void;
+  label?: string;
+}
+
+export const ProjectAccountSelector = ({
+  accounts,
+  selectedAddress,
+  onSelect,
+  label,
+}: ProjectAccountSelectorProps) => {
+  const { _ } = useLingui();
+  const [open, setOpen] = useState(false);
+  const rootRef = useClickOutside<HTMLDivElement>(() => {
+    if (open) setOpen(false);
+  });
+
+  const handleSelect = (address?: string) => {
+    if (!address) return;
+    onSelect(address);
+  };
+
+  const unnamedLabel = _(msg`Unnamed`);
+  const getName = (name?: string) => name?.trim() || unnamedLabel;
+  const selectedAccount = accounts.find(acc => acc.address === selectedAddress);
+
+  return (
+    <div ref={rootRef} className="relative w-full">
+      {label && (
+        <Body size="lg" className="text-bc-neutral-700 mb-[8px] font-bold">
+          {label}
+        </Body>
+      )}
+      {/* Button to toggle dropdown */}
+      <button
```

Comment:
couldn't have we just reuse an existing input component, eg inputs/new/SelectTextField, options label can be string or JSX.Element so they are customizable